src/components/application_manager/src/commands/mobile/show_request.cc

```diff
@@ -45,10 +45,46 @@ namespace commands {
 
 ShowRequest::ShowRequest(const MessageSharedPtr& message,
                          ApplicationManager& application_manager)
-    : CommandRequestImpl(message, application_manager) {}
+    : CommandRequestImpl(message, application_manager)
+    , core_result_code_(mobile_apis::Result::INVALID_ENUM) {}
 
 ShowRequest::~ShowRequest() {}
 
+void ShowRequest::HandleMetadata(const char* field_id,
+                                 int32_t field_index,
+                                 smart_objects::SmartObject& msg_params) {
+  smart_objects::SmartObject& metadata_struct =
+      (*message_)[strings::msg_params][strings::text_field_metadata];
+
+  if (metadata_struct.keyExists(field_id)) {
+    if (field_index != -1) {
+      msg_params[hmi_request::show_strings][field_index]
+                [hmi_request::field_types] =
+                    smart_objects::SmartObject(smart_objects::SmartType_Array);
+
+      const size_t num_tags = metadata_struct[field_id].length();
+      for (size_t i = 0; i < num_tags; ++i) {
+        const int32_t current_tag = metadata_struct[field_id][i].asInt();
+        msg_params[hmi_request::show_strings][field_index]
+                  [hmi_request::field_types][i] = current_tag;
+      }
+    } else {
+      LOG4CXX_INFO(logger_,
+                   "metadata tag provided with no item for "
+                       << field_id << ", ignoring with warning");
+      // tag provided with no item, ignore with warning
+      if (mobile_apis::Result::INVALID_ENUM == core_result_code_) {
+        core_result_code_ = mobile_apis::Result::IGNORED;
```

Comment:
The only issue I have with this is that the `IGNORED` result code seems to be only used when the entire RPC was ignored, rather than a specific part of it, and is generally considered a failure result code. It looks to me by the way it is used throughout the project, `WARNINGS` would be the correct result code to use in this case.

The descriptions of these result codes in the Mobile API seem to match up with this.
```
    <element name="IGNORED">
      <description>
        A command was ignored, because the intended result is already in effect.
        For example, SetMediaClockTimer was used to pause the media clock although the clock is paused already.
        NOTE: potentially replaces SUBSCRIBED_ALREADY
      </description>
    </element>
...
    <element name="WARNINGS">
      <description>The RPC (e.g. SubscribeVehicleData) executed successfully but one or more items have a warning or failure.</description>
    </element>
```